bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +827,258 @@ describe('ClientEncryption', function () {
+        });
+
+        expect(createDataKeySpy.callCount).to.equal(0);
+        const options = createCollectionSpy.getCall(0).args[1];
+        expect(options).to.deep.equal({ encryptedFields: { fields: 'not an array' } });
+      });
+    });
+
+    context('when options.encryptedFields.fields elements are', () => {
+      it('not objects they are passed along to createCollection', async () => {
+        const createCollectionSpy = sinon.spy(db, 'createCollection');
+        const keyId = new Binary(Buffer.alloc(16, 0));
+        const createDataKeyStub = sinon.stub(clientEncryption, 'createDataKey').resolves(keyId);
+        await clientEncryption.createEncryptedCollection(db, collectionName, {
+          createCollectionOptions: {
+            encryptedFields: { fields: ['not an array', { keyId: null }, { keyId: {} }] }
+          }
+        });
+
+        expect(createDataKeyStub.callCount).to.equal(1);
+        const options = createCollectionSpy.getCall(0).args[1];
+        expect(options).to.deep.equal({
+          encryptedFields: { fields: ['not an array', { keyId: keyId }, { keyId: {} }] }
+        });
+      });
+    });
+
+    it('only passes options.masterKey to createDataKey', async () => {
+      const masterKey = Symbol('key');
+      const createDataKey = sinon
+        .stub(clientEncryption, 'createDataKey')
+        .resolves(new Binary(Buffer.alloc(16, 0)));
+      const result = await clientEncryption.createEncryptedCollection(db, collectionName, {
+        provider: 'aws',
+        createCollectionOptions: { encryptedFields: { fields: [{}] } },
+        masterKey
+      });
+      expect(result).to.have.property('collection');
+      expect(createDataKey).to.have.been.calledOnceWithExactly('aws', { masterKey });
+    });
+
+    context('when createDataKey rejects', () => {
+      const customError = new Error('evil!');
+      const keyId = new Binary(Buffer.alloc(16, 0), 4);
+      const createCollectionOptions = {
+        encryptedFields: { fields: [{}, {}, { keyId: 'cool id!' }] }
+      };
+      const createDataKeyRejection = async () => {
+        const stub = sinon.stub(clientEncryption, 'createDataKey');
+        stub.onCall(0).resolves(keyId);
+        stub.onCall(1).rejects(customError);
+        stub.onCall(2).resolves(keyId);
+
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName, {
+            provider: 'local',
+            createCollectionOptions
+          })
+          .catch(error => error);
+
+        // At least make sure the function did not succeed
+        expect(error).to.be.instanceOf(Error);
+
+        return error;
+      };
+
+      it('throws MongoCryptCreateDataKeyError', async () => {
+        const error = await createDataKeyRejection();
+        expect(error).to.be.instanceOf(MongoCryptCreateDataKeyError);
+      });
+
+      it('thrown error has a cause set to the error that was thrown from createDataKey', async () => {
```

Comment:
I think this test case should be more specific - we want to test that the cause is the first error in the rejected errors array (since we decided that we're always returning the first error we encounter in the array of errors).

Reply:
Added case for first error